src/ebonite/ext/sqlalchemy/repository.py

```diff
@@ -116,13 +118,17 @@ def _create_object(self, object_type: Type[Attaching], obj: T, error_type) -> T:
             obj._id = p.id
             return obj
 
-    def _delete_object(self, object_type: Type[Attaching], obj, error_type):
+    def _delete_object(self, object_type: Type[Attaching], obj, ne_error_type, ie_error_type):
         with self._session() as s:
             p = s.query(object_type).filter(object_type.id == obj.id).first()
             if p is None:
-                raise error_type(obj)
+                raise ne_error_type(obj)
             logger.debug('Deleting object %s', p)
-            s.delete(p)
+            try:
+                s.delete(p)
+                s.commit()
```

Comment:
We could skip calling `commit` here as `_session` does it automatically. In `_create_object` this is needed to take auto-incrementing id assigned to object before finishing transaction.